exonum/src/runtime/mod.rs

```diff
@@ -253,10 +253,22 @@ pub trait Runtime: Send + fmt::Debug + 'static {
     /// and returned `Ok(())`. The results of the call (i.e., changes to the blockchain state)
     /// are guaranteed to be persisted from the call.
     ///
-    /// Note that a call to `start_adding_service()` may have happened indefinite time ago;
+    /// A call to `start_adding_service()` may have happened indefinite time ago;
     /// indeed, `commit_service()` is called for all services on the node startup. Likewise,
     /// `commit_service()` may be called an indefinite number of times for the same instance.
     ///
+    /// `snapshot` is the storage snapshot at the latest height when the method is called:
```

Comment:
This level of complexity is not quite comfortable, so it could make sense to remove the `snapshot` arg altogether (it isn't used by the Rust runtime). OTOH, I can imagine use cases where `snapshot` is useful (e.g., a potential "on node start" hook for Rust services).